cpp/ql/src/Critical/MissingCheckScanf.ql

```diff
@@ -100,7 +100,7 @@ private predicate fwdFlow(Instruction instr, ValueNumber vn) {
  */
 pragma[nomagic]
 predicate revFlow(Instruction instr, ValueNumber vn) {
-  fwdFlow(instr, vn) and
+  fwdFlow(instr, pragma[only_bind_out](vn)) and
```

Comment:
This way you get `(revFlow#delta minus isBarrier) join getASuccessor on succ`, whereas before it was  `(revFlow#delta minus isBarrier) join fwdFlow on vn`

Reply:
Yeah, my understanding was we wanted an early join with `getASuccessor`, but we were happy to let the optimizer pick what it joined it with (`fwdFlow` or `revFlow`).  As long as it isn't joining `fwdFlow` _with_ `revFlow` any more.

Reply:
Yeah, I can totally see why joining on `vn` would be bad. But is it not even better to _start_ with `fwdFlow`?

Reply:
... though if you can do it still with just one `pragma`, I will be happy with that. :)

Reply:
I'm just suggesting that we try the alternative join order and look at the tuple counts on the projects you've been checking today.

Reply:
Also, I was too impatient to run the whole query "before" due to how long it took, so I don't have before/after tuple counts :P But for example, quick-eval'ing `revFlow` went from 7 minutes down to 8 seconds.

Reply:
Totally understandable! I often also just post a partial "before" run (for example [here](https://github.com/github/codeql/pull/10261)) since it's often not even possible to fully evaluate the problematic pipeline.

Reply:
Was this alternative join order explored at all?